### PR TITLE
Makefile: add preliminary Makefile

### PR DESCRIPTION
**Pull request by Traneptora**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	$(CXX) -c -mf16c $(CXXFLAGS) -o $*.o $*.cpp
+
+$(AVX_OBJ):
+	$(CXX) -c -DUSE_FMA=0 -mavx $(CXXFLAGS) -o $*.o $*.cpp
+
+$(AVX2_OBJ):
+	$(CXX) -c -DUSE_FMA=1 -mavx2 -mfma $(CXXFLAGS) -o $*.o $*.cpp
+
+$(AVX512_OBJ):
+	$(CXX) -c -DUSE_FMA=1 -DZNEDI3_X86_AVX512 -mfma -mavx512f -mavx512cd -mavx512vl -mavx512bw -mavx512dq $(CXXFLAGS) -o $*.o $*.cpp
```
Comment (sekrit-twc) [inline Makefile:48]: Only -mavx512f is needed.

**Comment by Traneptora** [comment]
This PR was a misclick accident. Oops.

**Comment by Traneptora** [comment]
Should I fix and reopen?

**Comment by sekrit-twc** [comment]
This PR is not ready yet.

**Pull request closed without merging**